pkg/cmd/gptdocs.go

```diff
@@ -98,39 +98,71 @@ func (r GPTDocsRunner) Run(ctx context.Context) error {
 		}
 	}()
 
-	// Pick all events
-
-	var evt events.Event
-
 	// TODO: starting with syscall events only, but should also generate docs
 	//       for all other events if they don't exist. Note that for the other
 	//       events, the event definition should marshal everything into strings
 	//       so the chatGPT is able to understand the definition in order to
 	//       generate the event doc based on the template.
 
+	// Pick all events
+
+	var evt events.Event
+
 	allEvents := events.Definitions.Events()
 
+	// Check if the given events exist
+
+	for _, given := range r.GivenEvents {
+		_, ok := events.Definitions.GetID(given)
+		if !ok {
+			logger.Error("event definition not found", "event", given)
+		}
+	}
+
+	// Run all the events map through the GPT3 API
+
 	for _, evt = range allEvents {
 		if !evt.Syscall {
 			continue
 		}
+
+		// Check if the filename exists already and skip if it does
+
 		fileName := outputDirectory + "/" + evt.Name + ".md"
 		_, err := os.Stat(fileName)
 		if err == nil {
 			select {
 			case <-ctx.Done():
 				return nil
 			default:
+				logger.Debug("file already exists", "file", fileName)
 				continue
 			}
 		}
 
+		// Check if the event is in the given list of events and skip if it not
+
+		if len(r.GivenEvents) > 0 {
+			found := false
+			for _, given := range r.GivenEvents {
+				if strings.Contains(evt.Name, given) {
+					found = true
+				}
+			}
```

Comment:
should break when found?

Reply:
nope, this is correct. continue has to be outside this range loop.